src/main/kotlin/com/ajax/cryptocurrency/threads/PriceSaver.kt

```diff
@@ -53,13 +35,29 @@ data class PriceSaver(
                     val price: Float = jsonObject.getFloat("lprice")
 
                     val createdTimeStamp = OffsetDateTime.now(ZoneOffset.UTC)
-                    val cryptocurrency = Cryptocurrency(null, cryptocurrencyName = cryptocurrencyName, price = price, createdTime = createdTimeStamp.toLocalDateTime())
+                    val cryptocurrency = Cryptocurrency(
+                        null,
+                        cryptocurrencyName = cryptocurrencyName,
+                        price = price,
+                        createdTime = createdTimeStamp.toLocalDateTime()
+                    )
 
                     cryptocurrencyRepository.save(cryptocurrency)
                 }
             }.onFailure { e ->
                 logger.error("An exception occurred while saving prices", e)
             }
+            Thread.sleep(sleepTime.toLong())
+            executor.submit(this)
         }
     }
+
+    override fun run() {
+        logger.info("Started parsing: $cryptocurrencyName")
+        task.run()
+    }
```

Comment:
Why, instead of having two runnables, just not make one runnable?

Also question to @ajax-gnatiuk-s maybe use coroutines kotlin instead of java thread api?

Reply:
using of `coroutines` is up to author. I don't force using coroutines. We don't use coroutines in our prod services.